File: c-programming/DSA/recursion/towers-of-hanoi.py
def towers_of_hanoi(disks, src, dest, aux):
    """
    tower_of_hanoi problem

    :param disks: number of disks
    :param src: start needle
    :param dest: destination needle
    :param aux: auxiliary needle
    """
    if disks == 1:
        print_move(disks, src, dest)
    else:
        towers_of_hanoi(disks - 1, src, aux, dest)
        print_move(disks, src, aux)
        towers_of_hanoi(disks - 1, aux, dest, src)


def print_move(disk, src, dest):
    """
    prints the move

    :param disk: the disk moved
    :param src: source needle
    :param dest: destination needle
    """
    print(f"Move disk {disk} from needle {src} to needle {dest}")


if __name__ == '__main__':
    n = int(input("Number of disks: "))

    if (n):
        towers_of_hanoi(n, 'A', 'C', 'B')
    else:
        print("At least 1 disk is required")

File: DSA/recursion/towers-of-hanoi.c
#include <stdio.h>

void towers_of_hanoi(int disks, char src, char dest, char aux);

/**
 * main - recursion
 *
 * Return: 0
*/
int main(void)
{
	int num;

	printf("Number of disks: ");
	scanf("%d", &num);

	towers_of_hanoi(num, 'A', 'C', 'B');
}

/**
 * towers_of_hanoi - towers of hanoi
 *
 * @disks: number of disks
 * @src: starting peg
 * @dest: end peg (all pegs should end up here)
 * @aux: auxiliary peg
*/
void towers_of_hanoi(int disks, char src, char dest, char aux)
{
	if (disks == 1)
	{
		printf("Move disk 1 from Needle %c to Needle %c\n", src, dest);
		return;
	}
	/* move a disk on src to aux, using dest as auxiliary peg*/
	towers_of_hanoi(disks - 1, src, aux, dest);
	printf("Move disk %d from Needle %c to Needle %c\n", disks, src,
		   dest);
	/* move a disk on aux to dest, using src as auxiliary peg*/
	towers_of_hanoi(disks - 1, aux, dest, src);
}
